lib/src/Pos/Web/Server.hs

```diff
@@ -98,10 +106,60 @@ serveImpl app host port mWalletTLSParams mSettings =
     mySettings = setHost (fromString host) $
                  setPort (fromIntegral port) $
                  fromMaybe defaultSettings mSettings
-    mTlsConfig = tlsParamsToWai <$> mWalletTLSParams
+    mTlsConfig = tlsParamsToWai host port <$> mWalletTLSParams
 
-tlsParamsToWai :: TlsParams -> TLSSettings
-tlsParamsToWai TlsParams{..} = tlsSettingsChain tpCertPath [tpCaPath] tpKeyPath
+tlsParamsToWai
+    :: String -> Word16 -> TlsParams -> TLSSettings
+tlsParamsToWai host port TlsParams{..} = tlsSettingsWithCertCheck
+  where
+    tlsSettings =
+      tlsSettingsChain tpCertPath [tpCaPath] tpKeyPath
```

Comment:
I should have reviewed this one first, this answers my question about intermediates :)